### DISCUSSION TITLE
Ability To Make Connections Via Teddit

### DISCUSSION THREAD

**Discussion by seniorm0ment** [Docile-Alligator/Infinity-For-Reddit#897] [Ideas]
Requesting the ability to use Teddit (a privacy respective Reddit frontend) instead of Reddit for browsing. This would imply the user can optionally never connect to Reddit directly by using Teddit. This is especially nice for users who don't sign in.
There is one app, however it is very much in alpha and very restrictive as far as layout goes. The ability to stay on Infinity but make connections through Teddit would be amazing. 

https://codeberg.org/teddit/teddit
https://teddit.net/
Of course the ability for the user to choose a custom Teddit instance would be implied here. And of course, this would be optional for users who do not want it.

___________

Teddit's Features (Aimed at browsers, so some of these may be irrelevant to use in the app, however users would still not need to make a direct connection to Reddit if they used):

A free and open source alternative Reddit front-end focused on privacy. Inspired by the Nitter project.

No JavaScript or ads

All requests go through the backend, client never talks to Reddit

Prevents Reddit from tracking your IP or JavaScript fingerprint

Unofficial API (RSS & JSON support, no rate limits or Reddit account required)

Lightweight (teddit frontpage: ~30 HTTP requests with ~270 KB of data downloaded vs. Reddit frontpage: ~190 requests with ~24 MB)

Self-hostable. Anyone can setup an instance. An instance can either use Reddit's API with or without OAuth (so Reddit API key is not necessarily needed).

**Comment by r-a-y** [comment]
Teddit's unofficial API can be found here: https://codeberg.org/teddit/teddit/wiki#teddit-api

In particular, this is probably the endpoint you're interested in:

> In case you want the same JSON response what you get from the official Reddit API, you can get it from teddit API by adding query string &target=reddit. For example:
>
>    https://teddit.net/?api&target=reddit - return Reddit JSON for the frontpage

**Comment by yephny** [comment]
This option makes sense for anonymous browsing. It would switch back to reddit.com when the user chooses an account to log into.

**Comment by mardijker** [comment]
Appreciated if this can be implemented, as Reddit is blocked in my country, and for privacy reason as well.

**Comment by momobobe** [comment]
Libreddit is also an option, but maybe a little more complex to implement due to lack of api.

**Comment by ghost** [comment]
In the meantime: _Stealth_ already has this enhancement
https://gitlab.com/cosmosapps/stealth

**Comment by user451421541757324** [comment]
Teddit has now [more API endpoints](https://codeberg.org/teddit/teddit/pulls/373).

**Comment by highda** [comment]
Teddit is unmaintained and probably soon to be obsolete. Not to mention that Stealth's implementation is rather buggy.
LibReddit is discontinued and RedLib is the leading actively maintained front-end right now.
It indeed does lack an API and scraping sounds kinda meh, but definitely not impossible.
